src/sentry/eventstore/snuba/backend.py

```diff
@@ -201,6 +202,10 @@ def get_event_by_id(self, project_id, event_id, group_id=None):
                 selected_columns=["group_id"],
                 start=event.datetime,
                 end=event.datetime + timedelta(seconds=1),
+                # XXX: This is a hack to bust the snuba cache. We want to avoid the case where
+                # we cache an empty result, since this can result in us failing to fetch new events
+                # in some cases.
+                conditions=[["timestamp", ">", random.randint(0, 1000000000)]],
```

Comment:
Could we restrict this change to scenarios where `event.datetime` is > than now() - one hour ? So that we reduce the impact on events that have been in the DB for longer. 
Events technically could be backdated by a longer time range, though I suspect that would have a minimal impact.

Reply:
Good idea, I'll do that